graylog2-server/src/main/java/org/graylog2/shared/inputs/InputRegistry.java

```diff
@@ -124,6 +123,13 @@ public IOState<MessageInput> stop(MessageInput input) {
         return inputState;
     }
 
+    public void setup(IOState<MessageInput> inputState) {
+        final IOState<MessageInput> stopped = stop(inputState.getStoppable());
+        remove(stopped);
```

Comment:
From the UI, yes, because we can only enter SETUP from STOPPED. But using the API, you can transition from RUNNING to SETUP. In which case we need to stop the input.

Reply:
I see - quite right - fixed that.